frontend/app/mocks/node.ts

```diff
@@ -168,6 +213,48 @@ const handlers = [
       nameFr: preferredLanguageEntity.nameFr,
     });
   }),
+
+  /**
+   * Handler for GET requests to retrieve letters details.
+   */
+  http.get('https://api.example.com/letters?userId=', ({ params }) => {
+    const letterEntities = getLetterEntities(params.userId);
+
+    return HttpResponse.json(
+      letterEntities.map((letter) => {
+        return {
+          dateSent: letter.dateSent,
+          subject: letter.letterTypeCd,
+          referenceId: letter.referenceId,
+        };
+      }),
+    );
+  }),
+
+  /**
+   * Handler for GET requests to retrieve pdf
+   */
+  http.get('https://api.example.com/cct/letters/:referenceId', async ({ params }) => {
+    const encoder = new TextEncoder();
+    const pdfEntity = getPdfEntity(params.referenceId);
+    const stream = new ReadableStream({
+      start(controller) {
+        // Encode the string chunks using "TextEncoder".
+        controller.enqueue(encoder.encode('Gums'));
+        controller.enqueue(encoder.encode('n'));
+        controller.enqueue(encoder.encode('proses'));
+        controller.enqueue(encoder.encode(pdfEntity?.id));
+        controller.close();
+      },
+    });
+
+    // Send the mocked response immediately.
+    return new HttpResponse(stream, {
+      headers: {
+        'Content-Type': 'text/plain',
```

Comment:
I could be wrong about this, but shouldn't this be `application/octet-stream`?

Reply:
Sorry, I'm not sure I understand your point here.

This is mocking an API which is supposed to be returning a bytestream. I suppose we *could* designate the type as `application/pdf`, but definitely not `text/plain`.

If it's helpful, Box.com's file download API returns its bytestream as `application/octet-stream`: <https://developer.box.com/reference/get-files-id-content/#response>

Reply:
If the mock is returning a PDF then yes... but for not it's returning plain text content. Ok to switch it to `application/octet-stream` based on Box Platform.

Reply:
Use this header for now. Replace `{letter.referenceId}` is a placeholder and should be provider by the letter entity.
```
Content-Type: application/octet-stream
Content-Disposition: attachment; filename="{letter.referenceId}.pdf"
```

Reply:
Fixed: c20e06602530ac9a3fb8cf2b080a3b3fc36d0451